src/components/Calendar.tsx

```diff
@@ -0,0 +1,166 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { ZetkinEvent } from '../types/zetkin';
+import { Heading, Text } from '@adobe/react-spectrum';
+import { useEffect, useRef } from 'react';
+
+interface CalendarProps {
+    focusDate?: Date;
+    events: ZetkinEvent[];
+}
+
+const Calendar = ({ focusDate = new Date(Date.now()), events }: CalendarProps): JSX.Element => {
+    const calendar = useRef<HTMLDivElement>(null);
+    const calendarWrapper = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        const height = calendar.current?.clientHeight || 0;
+        const y = height / 24 * 8; // approx 8am
+        calendarWrapper.current?.scrollTo(0, y);
+    }, []);
+
+    const week = [
+        'misc.calendar.weeks.mon', 'misc.calendar.weeks.tue',
+        'misc.calendar.weeks.wed', 'misc.calendar.weeks.thu',
+        'misc.calendar.weeks.fri', 'misc.calendar.weeks.sat',
+        'misc.calendar.weeks.sun',
+    ];
+    const monday = new Date(new Date(focusDate)
+        .setDate(focusDate.getDate() - focusDate.getDay() + 1));
+    const nextMonday = new Date(new Date(monday)
+        .setDate(monday.getDate() + 7));
+
+    const eventsOfTheWeek = events.filter(event => {
+        return new Date(event.start_time) > monday &&
+            new Date(event.start_time) < nextMonday ||
+            new Date(event.end_time) > monday &&
+            new Date(event.end_time) < nextMonday;
+    });
+
+    const getEventsOfTheDay = (day: number) => {
+        if (day === 7) day = 0; // sunday has index 0 in the Date object
+        return eventsOfTheWeek.filter(event => (
+            new Date(event.start_time).getDay() === day ||
+            new Date(event.end_time).getDay() === day));
+    };
+
+    return (
+        <>
+            <div ref={ calendarWrapper } data-testid="calendar-wrapper" style={{
+                height: '100%',
+                overflow: 'scroll',
+                width: '100%',
+            }}>
+                <div style={{
+                    background: 'rgb(234, 234, 234)',
+                    display:'flex',
+                    gap:'1rem',
+                    justifyContent: 'space-between',
+                    position: 'sticky',
+                    top: 0,
+                    width: '100%',
+                    zIndex: 1,
+
+                }}>
+                    { week.map((day, index) => (
+                        <div key={ day } style={{
+                            alignItems: 'center',
+                            display:'flex',
+                            flexDirection: 'column',
+                            gap:'1rem',
+                            height: '100%',
+                            justifyContent: 'start',
+                            width: '100%',
+                        }}>
+                            <Heading level={ 3 }>
+                                <Msg id={ day }/>
```

Comment:
Another way of achieving this, without having to rely on strings from YAML, is to use the `<FormattedDate>` component. You can tell it to use the abbreviated day format, and ignore all other elements of a date, and that way just render the localized abbreviated weekday.